data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/query/DefaultBindableParametersStoredQuery.java

```diff
@@ -155,6 +156,10 @@ protected final void bindParameter(Binder binder,
                     }
                 }
             }
+        } else if (binding.getParameterBindingPath() != null && value instanceof Collection<?>) {
+            // For list of embedded values passed in criteria.
```

Comment:
This is case when value (non null) of `List<ProjectId>` for example is passed in criteria execution and then checking `binding.getParameterBindingPath()` should indicate this is embedded field/path but still not sure and would like if there is some better indicator to recognize this case.